### ISSUE TITLE
dialogs should restore focus to the previously focused element after closing

### ISSUE THREAD

**Issue by valdrinkoshi** [PolymerElements/iron-overlay-behavior#94]
_From @anthonytordillos on October 20, 2015 6:9_

"the object with focus before the dialog box is opened should be saved. This will allow restoring focus to this element when the dialog box is closed."

Is this something paper-dialog-behavior can provide, or are consumers of the element required to handle it themselves?

http://www.w3.org/TR/wai-aria-practices/#modal_dialog

_Copied from original issue: PolymerElements/paper-dialog-behavior#32_

**Comment by valdrinkoshi** [comment]
_From @anthonytordillos on November 27, 2015 23:7_

Ping?

**Comment by valdrinkoshi** [comment]
This is not currently implemented, but is a nice feature we could add: `paper-dialog-behavior` restores the focus to the origin element when closed.
@cdata WDYT? should this be the default behavior?

**Comment by valdrinkoshi** [comment]
_From @cdata on December 9, 2015 0:25_

This seems like a good feature to add. For reference, we do some wander `tabindex` stuff in `PaperMenuBehavior`. As an alternative approach to the one `PaperMenuBehavior` uses, we could manually update a property that keeps track of the last focused element in the dialog.

**Comment by valdrinkoshi** [comment]
@cdata, I cannot find `PaperMenuBehavior`, did you mean another behavior?

**Comment by valdrinkoshi** [comment]
_From @cdata on December 9, 2015 17:51_

Sorry, I meant https://github.com/PolymerElements/iron-menu-behavior :)
